packages/types/src/authentication.ts

```diff
@@ -32,24 +32,34 @@ export type SignUpSuccessfullResponseBody = z.infer<
     typeof SignUpSuccessfullResponseBody
 >;
 
-export const SignUpResponseBody = z
-    .object({
-        status: z.enum([
-            'UNAVAILABLE_EMAIL',
-            'UNAVAILABLE_NICKNAME',
-            'INVALID_EMAIL',
-            'WEAK_PASSWORD',
-        ]),
-    })
-    .or(SignUpSuccessfullResponseBody);
+export const SignUpFailureReasons = z.enum([
+    'UNAVAILABLE_NICKNAME',
+    'UNAVAILABLE_EMAIL',
+    'INVALID_EMAIL',
+    'WEAK_PASSWORD',
+]);
+
+export type SignUpFailureReasons = z.infer<typeof SignUpFailureReasons>;
+
+export const SignUpFailureResponseBody = z.object({
+    status: z.literal('FAILURE'),
+    signFailureReasons: SignUpFailureReasons.array(),
+});
+
+export type SignUpFailureResponseBody = z.infer<
+    typeof SignUpFailureResponseBody
+>;
```

Comment:
<!-- Praises highlight something positive. Try to leave at least one of these comments per review (if it exists :^)  -->
**👏 praise:** ‏Good job, this is great to handle errors that way! Back-end code done with quality in mind is interesting to read.